G/GCCBootstrap/build_tarballs.jl

```diff
@@ -0,0 +1,107 @@
+    #           "9e633fb278b411a90636cc1c4bf1ffddcc8b0d214f5bacd74bfcdaac81d6035e"),
+    FileSource("http://mirrors.kernel.org/gnu/glibc/glibc-ports-2.12.1.tar.bz2",
+               "3986c166d08c30b83c9cc7d972651550b548386d5a02e4e4c985d9832de83110"),
+    FileSource("https://github.com/madler/zlib/archive/refs/tags/v1.2.11.tar.gz",
+               "629380c90a77b964d896ed37163f5c3a34f6e6d897311f1df2a7016355c45eff",
+               filename="zlib-1.2.11.tar.gz"),
+    FileSource("http://mirrors.kernel.org/gnu/ncurses/ncurses-6.2.tar.gz",
+               "30306e0c76e0f9f1f0de987cf1c82a5c21e1ce6568b9227f7da5b71cbea86c9d"),
+    FileSource("http://mirrors.kernel.org/gnu/libiconv/libiconv-1.16.tar.gz",
+               "e6a1b1b589654277ee790cce3734f07876ac4ccfaecbee8afa0b649cf529cc04"),
+    FileSource("http://mirrors.kernel.org/gnu/gettext/gettext-0.21.tar.xz",
+               "d20fcbb537e02dcf1383197ba05bd0734ef7bf5db06bdb241eb69b7d16b73192"),
+    FileSource("http://mirrors.kernel.org/gnu/binutils/binutils-2.29.1.tar.xz",
+               "e7010a46969f9d3e53b650a518663f98a5dde3c3ae21b7d71e5e6803bc36b577"),
+    FileSource("http://mirrors.kernel.org/gnu/binutils/binutils-2.38.tar.xz",
+               "e316477a914f567eccc34d5d29785b8b0f5a10208d36bbacedcc39048ecfe024"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/crosstool-ng*/
+
+# These tools will help us to bootstrap
+apk add build-base texinfo help2man ncurses-dev
+
+# Copy in our extra patches for all packages
+for package in ${WORKSPACE}/srcdir/patches/*; do
+    package="$(basename "${package}")"
+    for version in ${WORKSPACE}/srcdir/patches/${package}/*; do
+        version="$(basename "${version}")"
+        if [ ! -d packages/${package}/${version} ]; then
+            continue
+        fi
+        cp -v ${WORKSPACE}/srcdir/patches/${package}/${version}/* packages/${package}/${version}/
+    done
+done
+
+# Disable some checks that ct-ng performs
+export CT_ALLOW_BUILD_AS_ROOT_SURE=1
+
+# Unset some things that BB automatically inserts into the environment,
+# but which crosstool-ng rightfully complains about.
+unset LD_LIBRARY_PATH
+for TOOL in CC CXX LD AS AR FC OBJCOPY OBJDUMP RANLIB STRIP LIPO MESON NM READELF; do
+    unset "${TOOL}" "BUILD_${TOOL}" "${TOOL}_BUILD" "${TOOL}_FOR_BUILD" "HOST${TOOL}"
+done
+PATH="$(printf "%s" "$(echo -n "${PATH}" | tr ':' '\n' | grep -v "/opt")" | tr '\n' ':')"
+
+
+# Build crosstool-ng for the current host
+./configure --enable-local
+make -j${nproc}
+
+# Generate the appropriate crosstool-ng config file for our current target
+${WORKSPACE}/srcdir/gen_config.sh > .config
+./ct-ng upgradeconfig
+./ct-ng build
+[[ -f "${bindir}/${bb_target}-gcc" ]]
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = filter(p -> Sys.islinux(p), supported_platforms())
```

Comment:
```suggestion
platforms = filter(Sys.islinux, supported_platforms())
```
Why only Linux?  What for the other systems?

Reply:
I will add Windows for sure, but I don't think FreeBSD or macOS are supported by crosstool-ng.  I may try to add support, but that's going to have to wait for a while.   See my overall comment about the purpose of this package for more details.